### PR TITLE
[routing-manager] derive on-link prefix from Extended PAN ID

### PR DESCRIPTION
**Pull request by jwhui**
To align with latest Thread Specification.

### CODE REVIEW

File: tests/scripts/thread-cert/border_router/test_vicarious_router_solicit.py [hunk 1]
```diff
@@ -82,13 +84,15 @@ class MultiThreadNetworks(thread_cert.TestCase):
...
             'router_selection_jitter': 1,
+            'extended_panid': '08090a0b0c0d0e0f'
         },
         ROUTER2: {
             'name': 'Router_2',
             'allowlist': [BR2],
             'version': '1.2',
             'channel': CHANNEL2,
             'router_selection_jitter': 1,
+            'extended_panid': '08090a0b0c0d0e0f'
```
Comment (simonlingoogle) [inline tests/scripts/thread-cert/border_router/test_vicarious_router_solicit.py:95]: Is it required for this test to use a specified extpanid ?

File: src/core/border_router/routing_manager.cpp [hunk 2]
```diff
@@ -397,6 +379,24 @@ void RoutingManager::HandleNotifierEvents(Events aEvents)
         StartRoutingPolicyEvaluationJitter(kRoutingPolicyEvaluationJitter);
     }
 
+    if (aEvents.Contains(kEventThreadExtPanIdChanged))
```
Comment (simonlingoogle) [inline src/core/border_router/routing_manager.cpp:382]: Should we add a test for this logic?

**Review by simonlingoogle — APPROVED** [review]
LGTM 👍 

with nits

File: tests/scripts/thread-cert/border_router/test_vicarious_router_solicit.py (hunk 1 above)
Comment (wgtdkp) [reply tests/scripts/thread-cert/border_router/test_vicarious_router_solicit.py:95]: Test cases in this file are assuming different on-link prefixes in the two Thread networks. I think some assertions could fail if the same extended pan id is used.

**Review by wgtdkp — APPROVED** [review]
Thanks! LGTM 👍

File: tests/scripts/thread-cert/border_router/test_vicarious_router_solicit.py (hunk 1 above)
Comment (simonlingoogle) [reply tests/scripts/thread-cert/border_router/test_vicarious_router_solicit.py:95]: SGTM. 

I guess the current on-link prefix mechanism works even if two different Thread networks are sharing the same extended panid. The on-link prefix would be same no matter which BR is advertising it.

Comment (wgtdkp) [reply tests/scripts/thread-cert/border_router/test_vicarious_router_solicit.py:95]: Yeah, just the test assertions are assuming different prefixes.

File: src/core/border_router/routing_manager.cpp [hunk 3]
```diff
@@ -222,37 +223,18 @@ void RoutingManager::GenerateNat64Prefix(void)
 }
 #endif
 
-Error RoutingManager::LoadOrGenerateRandomOnLinkPrefix(void)
+void RoutingManager::GenerateOnLinkPrefix(void)
 {
-    Error error     = kErrorNone;
-    bool  generated = false;
-
-    if (Get<Settings>().Read<Settings::OnLinkPrefix>(mLocalOnLinkPrefix) != kErrorNone ||
```
Comment (abtink) [inline src/core/border_router/routing_manager.cpp:230]: I think we can also deprecate this settings (maybe in a next PR)?

**Review by abtink — APPROVED** [review]
LGTM. 👍

File: src/core/border_router/routing_manager.cpp (hunk 3 above)
Comment (jwhui) [reply src/core/border_router/routing_manager.cpp:230]: Will do in a subsequent PR.

File: src/core/border_router/routing_manager.cpp (hunk 2 above)
Comment (jwhui) [reply src/core/border_router/routing_manager.cpp:382]: Added a test - see below.

**Pull request merged**